include/deal.II/particles/particle_accessor.h

```diff
@@ -0,0 +1,210 @@
+      * whether this is a valid location in the simulation domain.
+      *
+      * @param [in] new_location The new location for this particle.
+      */
+    void
+    set_location (const Point<spacedim> &new_location);
+
+    /**
+     * Get the location of this particle.
+     *
+     * @return The location of this particle.
+     */
+    const Point<spacedim> &
+    get_location () const;
+
+    /**
+     * Set the reference location of this particle.
+     *
+     * @param [in] new_reference_location The new reference location for
+     * this particle.
+     */
+    void
+    set_reference_location (const Point<dim> &new_reference_location);
+
+    /**
+     * Return the reference location of this particle in its current cell.
+     */
+    const Point<dim> &
+    get_reference_location () const;
+
+    /**
+     * Return the ID number of this particle.
+     */
+    types::particle_index
+    get_id () const;
+
+    /**
+     * Tell the particle where to store its properties (even if it does not
+     * own properties). Usually this is only done once per particle, but
+     * since the particle generator does not know about the properties
+     * we want to do it not at construction time. Another use for this
+     * function is after particle transfer to a new process.
+     */
+    void
+    set_property_pool(PropertyPool &property_pool);
+
+    /**
+      * Returns whether this particle has a valid property pool and a valid
+      * handle to properties.
+      */
+    bool
+    has_properties () const;
+
+    /**
+     * Set the properties of this particle.
+     *
+     * @param [in] new_properties A vector containing the
+     * new properties for this particle.
+     */
+    void
+    set_properties (const std::vector<double> &new_properties);
+
+    /**
+     * Get write-access to properties of this particle.
+     *
+     * @return An ArrayView of the properties of this particle.
+     */
+    const ArrayView<double>
+    get_properties ();
+
+    /**
+     * Get read-access to properties of this particle.
+     *
+     * @return An ArrayView of the properties of this particle.
+     */
+    const ArrayView<const double>
+    get_properties () const;
+
+    /**
+     * Returns the size in bytes this particle occupies if all of its data is
+     * serialized (i.e. the number of bytes that is written by the write_data
+     * function of this class).
+     */
+    std::size_t
+    serialized_size_in_bytes() const;
+
+    /**
+     * Get a cell iterator to the cell surrounding the current particle.
+     * As particles are organized in the structure of a triangulation,
+     * but the triangulation itself is not stored in the particle this
+     * operation requires a reference to the triangulation.
+     */
+    typename parallel::distributed::Triangulation<dim,spacedim>::cell_iterator
+    get_surrounding_cell (const parallel::distributed::Triangulation<dim,spacedim> &triangulation) const;
```

Comment:
why does it have to be `p::d::Tria` and not `p::Tria` or even `Tria`?

Reply:
It probably doesn't: `p::d::Tria` is derived from `::Tria`, so if the argument was a reference to the latter, everything would continue to work. Furthermore, the cell iterators of the two classes are the same type, so the function could just return a `::Tria::cell_iterator` without harm.